include/grpcpp/server_impl.h

```diff
@@ -199,6 +199,16 @@ class Server : public grpc::ServerInterface, private grpc::GrpcLibraryCodegen {
 
   grpc_server* server() override { return server_; }
 
+ protected:
+  void set_health_check_service(
+      std::unique_ptr<grpc::HealthCheckServiceInterface> service) {
+    health_check_service_ = std::move(service);
+  }
+
+  bool health_check_service_disabled() {
+    return health_check_service_disabled_;
+  }
+
```

Comment:
Where are the tests or sample usages for the new methods?

Reply:
They're internal only.

Reply:
Can we get a comment then saying that these are for internal use only and not API. Because protected is normally considered API since anyone can derive a class based on this.

Reply:
They're certainly allowed to be used externally. It's just that right now, the usages are internal

Reply:
Do you want some external testes for these? They are just getters and setters and I have internal tests that use these.

Reply:
If they're allowed to be used externally, then this is an API addition, which requires a gRFC.

I think it's better to mark them as being for internal use only.

Reply:
Added comment, PTAL.

Reply:
@vjpai As suggested offline, moved the comments to above each function. PTAL.